src/lib.rs

```diff
@@ -1,115 +1,4 @@
-#![allow(clippy::module_inception)]
-#![allow(clippy::mutable_key_type)]
+pub mod core;
```

Comment:
Let't not make `core` pub. We should only expose components that are needed. All core component should be available at root for eg `core::Foo` and not `core::foo::Foo`.

Reply:
Make sure we don't use `*` to expose everything.